src/Workspaces/Core/Portable/CodeActions/CodeActionWithOptions.cs

```diff
@@ -50,11 +51,11 @@ public abstract class CodeActionWithOptions : CodeAction
             return operations;
         }
 
-        internal override async Task<ImmutableArray<CodeActionOperation>> GetOperationsCoreAsync(
-            Solution originalSolution, IProgressTracker progressTracker, CancellationToken cancellationToken)
+        private protected sealed override async Task<ImmutableArray<CodeActionOperation>> GetOperationsCoreAsync(
+            Solution originalSolution, IProgress<CodeAnalysisProgress> progress, CancellationToken cancellationToken)
```

Comment:
this is a method that would become purely protected once we agree that thsi api shape looks good.